moving_state_to_application_state : Moves component level state to application level state(possible solution)

paschalidi opened the pull request:
Hey again,

> moving_state_to_application_state: comments possibe solution	


Your comment
> Move the state from component_card.js to the application state
was not really clear to me but I think I figured out what you would like me to do.

Check the commit and give me some feedback please so that I can implement that tomorrow.

Commit:
moving_state_to_application_state: comments possibe solution

MentalAtom commented on src/components/component_card.js:
```diff
@@ -5,7 +5,10 @@ import * as actions from '../actions';
+    // move this whole state to a reducer and trigger actions accordingly
+    // this way you will keep the logic of the application only to
+    // the level of the reducers and not the component level.
+    
```
You should consider what does and does not need to be in the state (e.g. what changes?)

For anything that does change, try and move this into the main application state so it could be handled by any component, not just this one - based on the contents of the store. This makes the whole thing much more generic and less likely to break.

There should be nothing left in this individual component state when you are done.

paschalidi replied:
Hey
I came up with an implementation.
Please feel free to give me some additional feedback if you find something obscure. 
From my side, I want to mention that I took a close look at immutability and reusability also (at least in my understanding) the code is clean and elegant.

Please let me know what you think @MentalAtom

Commit:
moving_state_to_application_state: creates action and reducer for update

Commit:
moving_state_to_application_state: refactors component card and content

this way the state is comming from redux and it is not a local state. As MentalAtom mentiones in his comment "This makes the whole thing much more generic and less likely to break."

Commit:
moving_state_to_application_state: adds testing
